video/input.h

```diff
@@ -70,26 +74,69 @@ class Input
 
     virtual const std::string getCameraName() const
     {
-        return "unknown_camera";
+        return DefaultCameraName;
     }
 
-    virtual bool readFrame(cv::Mat &outFrame)
+    virtual cv::Size getOutputSize() const
     {
-        return false;
+        return cv::Size();
     }
 
-    virtual cv::Size getOutputSize() const
+    virtual bool needsUnwrapping()
     {
-        return cv::Size();
+        // only panoramic cameras are defined with the camera name specified
+        return getCameraName() != DefaultCameraName;
     }
 
+    virtual bool readFrame(cv::Mat &outFrame) = 0;
+
+    virtual void setOutputSize(const cv::Size &outSize)
+    {}
+
 protected:
     /*
      * Default constructor is protected to stop users directly creating objects
      * of this class.
      */
     Input()
     {}
+
+private:
+    std::unique_ptr<std::thread> m_ImageThread;
```

Comment:
You don't need to make pointers to ``std::thread``. Just default construct it (i.e. to an invalid thread) and then assign it i.e. ``m_ImageThread = std::thread([this](Net::Node *n){ runImageSink(n); }, &node));``